src/message.rs

```diff
@@ -1762,7 +1766,15 @@ pub async fn markseen_msgs(context: &Context, msg_ids: Vec<MsgId>) -> Result<()>
         _curr_ephemeral_timer,
     ) in msgs
     {
-        if curr_state == MessageState::InFresh || curr_state == MessageState::InNoticed {
+        if curr_state == MessageState::InFresh
+            && matches!(
+                curr_download_state,
+                DownloadState::Available | DownloadState::InProgress
```

Comment:
What about `DownloadState::Failed`? I'm not sure when `DownloadState::Failed` is set, but IIUC it's e.g. when the network vanishes while trying to download a message, and in this case, we still shouldn't mark it as `\Seen`.

Reply:
Fixed this and extended the test accordingly. Only `DownloadedState::Done` messages must be upgraded to `InSeen`, the rest -- only to `InNoticed`